### PR TITLE
fix readme to point to 0.5.1

### PR DESCRIPTION
**Pull request by nlopezgi**
https://github.com/bazelbuild/rules_docker/issues/500
- Also adds missing skylib declarations for providers.bz
- also remove unnecessary setting of runfiles tree in java image

### CODE REVIEW

File: testdata/BUILD [hunk 1]
```diff
@@ -25,6 +25,8 @@ load("//testdata:utils.bzl", "generate_deb")
 
 exports_files(["pause.tar"])
 
+exports_files(["BUILD"])
```
Comment (xingao267) [inline testdata/BUILD]: why is this needed?

Comment (nlopezgi) [reply testdata/BUILD]: Because py_image is using the BUILD file it its data attr

**Comment by zaneclaes** [comment]
@nlopezgi this PR contains invalid syntax and the README is no longer usable at all.

File: README.md [hunk 2]
```diff
@@ -86,9 +86,9 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
 http_archive(
     name = "io_bazel_rules_docker",
-    sha256 = "5d64414477ffe87b16e248986c0731aee821ea3d626121cfeeec4ef7096812ca",
-    strip_prefix = "rules_docker-0.5.0",
-    urls = ["https://github.com/bazelbuild/rules_docker/archive/v0.5.0.tar.gz"],
+    sha256 = "29d109605e0d6f9c892584f07275b8c9260803bf0c6fcb7de2623b2bedc910bd",
+    strip_prefix = "rules_docker-0.5.1,
```
Comment (zaneclaes) [inline README.md]: Missing closing quotation.
